festim/boundary_conditions/flux_bc.py

```diff
@@ -0,0 +1,258 @@
+        >>> ParticleFluxBC(subdomain=my_subdomain, value=lambda x, t: 1 + x[0] + t, species="H")
+    """
+
+    def __init__(self, subdomain, value, species):
+        super().__init__(subdomain=subdomain, value=value)
+        self.species = species
+
+        self.value_fenics = None
+        self.bc_expr = None
+
+    @property
+    def temperature_dependent(self):
+        if self.value is None:
+            return False
+        if isinstance(self.value, fem.Constant):
+            return False
+        if callable(self.value):
+            arguments = self.value.__code__.co_varnames
+            return "T" in arguments
+        else:
+            return False
+
+    def create_value_fenics(
+        self, mesh, function_space: fem.FunctionSpace, temperature, t: fem.Constant
+    ):
+        """Creates the value of the boundary condition as a fenics object and sets it to
+        self.value_fenics.
+        If the value is a constant, it is converted to a fenics.Constant.
+        If the value is a function of t, it is converted to a fenics.Constant.
+        Otherwise, it is converted to a fenics.Function and the
+        expression of the function is stored in self.bc_expr.
+
+        Args:
+            mesh (dolfinx.mesh.Mesh) : the mesh
+            function_space (dolfinx.fem.FunctionSpace): the function space
+            temperature (float): the temperature
+            t (dolfinx.fem.Constant): the time
+        """
+        x = ufl.SpatialCoordinate(mesh)
+
+        if isinstance(self.value, (int, float)):
+            self.value_fenics = F.as_fenics_constant(mesh=mesh, value=self.value)
+
+        elif callable(self.value):
+            arguments = self.value.__code__.co_varnames
+
+            if "t" in arguments and "x" not in arguments and "T" not in arguments:
+                # only t is an argument
+                if not isinstance(self.value(t=float(t)), (float, int)):
+                    raise ValueError(
+                        f"self.value should return a float or an int, not {type(self.value(t=float(t)))} "
+                    )
+                self.value_fenics = F.as_fenics_constant(
+                    mesh=mesh, value=self.value(t=float(t))
+                )
+            else:
+                self.value_fenics = fem.Function(function_space)
+                kwargs = {}
+                if "t" in arguments:
+                    kwargs["t"] = t
+                if "x" in arguments:
+                    kwargs["x"] = x
+                if "T" in arguments:
+                    kwargs["T"] = temperature
+
+                # store the expression of the boundary condition
+                # to update the value_fenics later
+                self.bc_expr = fem.Expression(
+                    self.value(**kwargs),
+                    function_space.element.interpolation_points(),
+                )
+                self.value_fenics.interpolate(self.bc_expr)
+
+
+class HeatFluxBC(FluxBCBase):
```

Comment:
With this implementation, can the HeatFluxBC be temperature-dependent?
I don't think so. Users should be able to do:

```python
my_flux = HeatFluxBc(lambda T: h * (T-T_ext))
```

Reply:
With this change, I think that a lot of the code will be duplicated from `ParticleFlux` therefore all this functionality can be moved to `FluxBCBase`

Reply:
Ah of course yes

Reply:
oh wow that's a lot of code that went to FluxBCBase haha